test/NuGet.Core.Tests/NuGet.PackageManagement.Test/BuildIntegratedTests.cs

```diff
@@ -1562,191 +1618,239 @@ await nuGetPackageManager.UninstallPackageAsync(buildIntegratedProject, packageI
+            var packageSpec = ProjectTestHelpers.GetPackageSpec(
+                testSettings,
+                projectName,
+                pathContext.SolutionRoot,
+                framework: "net452");
 
-                var message = string.Empty;
+            var msBuildNuGetProjectSystem = new TestMSBuildNuGetProjectSystem(
+                projectTargetFramework,
+                testNuGetProjectContext,
+                projectFullPath: Path.GetDirectoryName(packageSpec.FilePath),
+                projectName);
 
-                await nuGetPackageManager.InstallPackageAsync(buildIntegratedProject, packageIdentity, new ResolutionContext(), new TestNuGetProjectContext(),
-                        sourceRepositoryProvider.GetRepositories(), sourceRepositoryProvider.GetRepositories(), CancellationToken.None);
+            var buildIntegratedProject = new TestPackageReferenceNuGetProject(packageSpec, msBuildNuGetProjectSystem);
+            testSolutionManager.NuGetProjects.Add(buildIntegratedProject);
 
-                buildIntegratedProject.ExecuteInitScriptAsyncCalls.Clear();
+            await nuGetPackageManager.InstallPackageAsync(buildIntegratedProject, packageIdentity, new ResolutionContext(), new TestNuGetProjectContext(),
+                    sourceRepositoryProvider.GetRepositories(), sourceRepositoryProvider.GetRepositories(), CancellationToken.None);
 
-                // Act
-                await nuGetPackageManager.UninstallPackageAsync(buildIntegratedProject, packageIdentity.Id,
-                    new UninstallationContext(), new TestNuGetProjectContext(), token);
+            buildIntegratedProject.ExecuteInitScriptAsyncCalls.Clear();
 
-                // Assert
-                Assert.Equal(0, buildIntegratedProject.ExecuteInitScriptAsyncCalls.Count);
-            }
+            // Act
+            await nuGetPackageManager.UninstallPackageAsync(buildIntegratedProject, packageIdentity.Id,
+                new UninstallationContext(), new TestNuGetProjectContext(), token);
+
+            // Assert
+            buildIntegratedProject.ExecuteInitScriptAsyncCalls.Count.Should().Be(0);
         }
 
-        [Fact(Skip = "https://github.com/NuGet/Home/issues/10212")]
+        [Fact]
         public async Task UpdatePackageCallsInitPs1OnNewPackages()
         {
             // Arrange
+            using var pathContext = new SimpleTestPathContext();
+            using var testSolutionManager = new TestSolutionManager(pathContext);
+
+            string projectName = "TestProjectName";
             var packageIdentity = new PackageIdentity("nuget.core", NuGetVersion.Parse("2.8.3"));
             var updateIdentity = new PackageIdentity("nuget.core", NuGetVersion.Parse("2.8.5"));
-            var dependencyIdentity = new PackageIdentity("Microsoft.Web.Xdt", NuGetVersion.Parse("2.1.0"));
             var sourceRepositoryProvider = TestSourceRepositoryUtility.CreateV2OnlySourceRepositoryProvider();
```

Comment:
Similar feedback. Create a local provider, and create the packages used in the test locally.